### PR TITLE
Feature/ub bv 1084

### PR DESCRIPTION
**Pull request by ubozov**
These commits contains the following changes:

1. imports product to `dappctrl`
2. install/remove products
3. flag to display log into console output
4. fix `gcc` compile error on Windows
5. database is created by `dappctrl db-create`

Resolves BV-1084, BV-1085.

### CODE REVIEW

File: product/misc.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	productDir     = "product"
+	envFile        = ".env.config.json"
+	configFile     = "install.config.json"
```
Comment (drew2a) [inline product/misc.go]: May be to move `envFile` and `configFile` to the config?

File: product/proc.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+	offeringTemplate = "offering.json"
+	accessTemplate   = "access.json"
+
+	serverProduct = "server.json"
+	clientProduct = "client.json"
+
+	adapterConfig       = "adapter.config.json"
+	agentAdapterConfig  = "adapter.agent.config.json"
+	clientAdapterConfig = "adapter.client.config.json"
```
Comment (drew2a) [inline product/proc.go]: I prefer to see all this variables in the config, also

File: product/product.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+		if installed {
+			_, err := run(role, productPath, "remove")
+			if err != nil {
+				return err
+			}
+
+			if imported {
+				err := writeVariable(envPath, productInstall, false)
+				if err != nil {
```
Comment (drew2a) [inline product/product.go]: This is too nested, from my perspective.

File: command/funcs.go [hunk 4]
```diff
@@ -48,6 +49,8 @@ func processedCommonFlags(d *dapp.Dapp, help string) error {
 	path := flag.String("workdir", "", "Dapp install directory")
 	src := flag.String("source", "", "Dapp install source")
 
+	flag.Bool("output", false, "Display log to console output")
```
Comment (furkhat) [inline command/funcs.go]: `verbose` is a better name. Many other tools use it to indicate what we need.

Comment (ubozov) [reply command/funcs.go]: updated

File: product/misc.go (hunk 1 above)
Comment (ubozov) [reply product/misc.go]: it's impossible
`envFile` is generated during product installation,
`config` is distributed with the product

File: product/proc.go (hunk 2 above)
Comment (ubozov) [reply product/proc.go]: these codes were moved from `dapp-openvpn/installer`

File: product/product.go (hunk 3 above)
Comment (ubozov) [reply product/product.go]: updated

File: product/proc.go [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+}
+
+func products(serverFile, clientFile, templateID string) (srvProduct,
+	cliProduct *data.Product, err error) {
+	srvProduct, err = productFromFile(serverFile)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if !productConcord(srvProduct, templateID) {
```
[the remark this answers is not in the document]
Comment (ubozov) [reply product/proc.go]: these codes were moved from `dapp-openvpn/installer/proc.go`

File: windows/service.go [hunk 6]
```diff
@@ -1,12 +1,11 @@
 // +build windows
 
-//go:generate goversioninfo -o ../resource.syso
+//go:generate goversioninfo -64 -o ../resource_windows_amd64.syso
```
[the remark this answers is not in the document]
Comment (ubozov) [reply windows/service.go]: updated

**Pull request merged**